File: Arrays/One Dimensional Arrays/removeDuplicates.cpp
#include <bits/stdc++.h>
using namespace std;

void removeDuplicates(char arr[]){

    int l = strlen(arr);
    if ((l == 0) or (l == 1)){
        return ;
    }
    int prev = 0;
    for (int current = 1; current < l; current++){
        if (arr[current] != arr[prev]){
            prev++;
            arr[prev] = arr[current];
        }

    }
    arr[prev + 1] = '\0';
    return ;
}


int main(){
    char arr[100];
    cin.getline(arr, 100);

    removeDuplicates(arr);
    cout << arr << endl;
    return 0;
}

File: Arrays/One Dimensional Arrays/removeDuplicates.py
arr = ['h', 'e', 'l', 'l', 'o']


def removeDuplicates(arr):

    l = len(arr)
    if ((l == 0) or (l == 1)):
        return

    prev = 0
    for curr in range(1, l):
        if (arr[curr] != arr[prev]):
            prev += 1
            arr[prev] = arr[curr]
    arr[prev + 1] = '\0'
    del(arr[prev + 1])
    return

removeDuplicates(arr)
print(arr)
